distribution/lib/Standard/Database/0.0.0-dev/src/DB_Table.enso

```diff
@@ -539,10 +541,21 @@ type DB_Table
               table.rename_columns (Map.from_vector [["name=(.*)".to_regex, "key:$1"]])
     @column_map Widget_Helpers.make_rename_name_vector_selector
     rename_columns : Map (Text | Integer | Regex) Text | Vector Text | Vector Vector -> Case_Sensitivity -> Boolean -> Problem_Behavior -> DB_Table ! Missing_Input_Columns | Ambiguous_Column_Rename | Too_Many_Column_Names_Provided | Invalid_Column_Names | Duplicate_Output_Column_Names
-    rename_columns self (column_map:(Map | Vector)=["Column"]) (case_sensitivity:Case_Sensitivity=Case_Sensitivity.Default) (error_on_missing_columns:Boolean=True) (on_problems:Problem_Behavior=Report_Warning) =
-        new_names = Table_Helpers.rename_columns self.column_naming_helper self.internal_columns column_map case_sensitivity error_on_missing_columns on_problems
-        Warning.with_suspended new_names names->
-            self.updated_columns (self.internal_columns.map c-> c.rename (names.at c.name))
+    rename_columns self (column_map:(DB_Table | Table | Map | Vector)=["Column"]) (case_sensitivity:Case_Sensitivity=Case_Sensitivity.Default) (error_on_missing_columns:Boolean=True) (on_problems:Problem_Behavior=Report_Warning) = case column_map of
+        _ : DB_Table -> Error.throw (Illegal_Argument.Error "column_map cannot be an in-database, use read to materialise.")
```

Comment:
We check for `DB_Table` here, but not in the `Table` variant.

I know, because we don't "know" the `DB_Table` in the `Standard.Table` library. Still the behaviour should be friendly in that case also. What is it currently? We should have a test (`in_memory_table.rename_columns db_table`) to check the behaviour is sane.

Reply:
I think we can still get a nice error if we trigger a `Table.from` conversion, because we have the `Table.from (that:DB_Table)` conversion that throws the error telling to materialize the table manually.

So we could try `Table.from` here and catch a `No_Such_Conversion` panic, falling back to the non-table case, but propagating any dataflow errors (the DB_Table case).